src/main/java/teammates/storage/api/CommentsDb.java

```diff
@@ -386,10 +386,7 @@ public void deleteCommentsByInstructorEmail(String courseId, String email) {
         Assumption.assertNotNull(Const.StatusCodes.DBLEVEL_NULL_INPUT, courseId);
         Assumption.assertNotNull(Const.StatusCodes.DBLEVEL_NULL_INPUT, email);
 
-        List<Comment> giverComments = this.getCommentEntitiesForGiver(courseId, email);
-
-        getPm().deletePersistentAll(giverComments);
-
+        getCommentsForGiverQuery(courseId, email).deletePersistentAll();
         getPm().flush();
```

Comment:
Hmm I think there might not be a need actually, as `Query.deletePersistentAll` [already flushes](https://db.apache.org/jdo/api20/apidocs/javax/jdo/Query.html#deletePersistentAll()). Thanks!